src/harness/sas_objects.py

```diff
@@ -0,0 +1,353 @@
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import json
+import os
+import sas
+import urllib
+
+
+class Grant:
+
+    """
+    A Grant object has the following methods
+     __init__
+          Parameters: grant ID, grant request
+    Retrieve  grant ID,
+    Retrieve  grant request
+    Retrieve operation params
+    Construct heartbeat request
+    Update was authorized in last heartbeat? state
+    Retrieve was authorized in last heartbeat
+    """
+
+    def __init__(self, g_id, g_request):
+        self.grant_id = g_id
+        self.grant_request = g_request
+        self.grant_authorized_in_last_heartbeat = False
+
+    def get_grant_id(self):
+        """
+
+        :rtype: grant_id
+        """
+        return self.grant_id
+
+    def get_grant_request(self):
+        """
+
+        :rtype: grant_request
+        """
+        return self.grant_request
+
+    def get_request_operation_param(self):
+        """
+
+        :rtype: operationParam
+        """
+        return self.grant_request['grantRequest'][0]['operationParam']
+
+    def construct_heartbeat_request(self):
+        """
+
+        :rtype: heartbeat_request
+        """
+        heartbeat_request = {
+            'heartbeatRequest': [{
+                'grantId': self.grant_id,
+                'cbsdId': self.grant_request['grantRequest'][0]['cbsdId'],
+                'operationState': "GRANTED"
+            }]
+        }
+        return heartbeat_request
+
+    def update_grant_authorized_in_last_heartbeat(self, is_authorized):
+        self.grant_authorized_in_last_heartbeat = is_authorized
+
+    def get_grant_authorized_in_last_heartbeat(self):
+        """
+
+        :rtype: grant_authorized_in_last_heartbeat
+        """
+        return self.grant_authorized_in_last_heartbeat
+
+
+class Cbsd:
+
+    """
+    A CBSD object has the following methods
+        __init__
+            Parameters: CBSD ID, successful registration request, grant ID, successful grant request
+        Retrieve   CBSD ID,
+        Retrieve successful registration request,
+        Retrieve grant request, successful
+            Parameter: grant request
+        Does this CBSD have at least one active grant?
+        Construct heartbeat request for all active grants
+        Return operating params for all grants which are authorized to transmit
+    """
+    def __init__(self, c_id, reg_request, g_id, grant_requests):
```

Comment:
Please spell out words rather than using non-standard abbreviations.

Reply:
Abbreviations   are changed to meaningful names